cpp/src/arrow/util/byte_stream_split_internal.h

```diff
@@ -584,52 +446,63 @@ inline void ByteStreamSplitDecodeScalarDynamic(const uint8_t* data, int width,
   DoMergeStreams(src_streams.data(), width, num_values, out);
 }
 
+template <int kNumStreams>
+void ByteStreamSplitDecodeSimdDispatch(const uint8_t* data, int width, int64_t num_values,
+                                       int64_t stride, uint8_t* out);
+
+extern template void ByteStreamSplitDecodeSimdDispatch<2>(const uint8_t*, int, int64_t,
```

Comment:
You may need `ARROW_TEMPLATE_EXPORT` to fix the Windows linking errors (see CI builds).